frappe/core/doctype/doctype/doctype.py

```diff
@@ -1418,10 +1415,9 @@ def validate_data_field_type(docfield):
 				)
 				df_options_str = "<ul><li>" + "</li><li>".join(_(x) for x in data_field_options) + "</ul>"
 
-				frappe.msgprint(text_str + df_options_str, title="Invalid Data Field", raise_exception=True)
+				frappe.msgprint(text_str + df_options_str, title="Invalid Data Field")
```

Comment:
If not raising, is there a point to have this msgprint at all?

Reply:
@surajshetty3416 in most cases it wont make sense, but other apps can potentially modify behaviour based on options like @gavindsouza mentioned in the review above. 

Vast majority of users wont need it hence kept the message :eyes:

Reply:
A standard way to extend is to update `frappe.model.data_field_options`. 
So raising still can be relevant since we are checking `frappe.model.data_field_options` for valid options.

---

Better to avoid raising only during migration?